wireapp/kalium
# chore: add script for generating new API version #WPB-11975

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11975" title="WPB-11975" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-11975</a>  [Android] Adopt API v7
  </td></table>
  <br />
 
















# What's new in this PR?

### Issues

While implementing API v7 I noticed that we need to add many new classes in a repetitive way. Doing it manual is prone to errors.

### Solutions

Added script that automates the process. Added comments about the script in a likely places that develop can start looking in.

### Testing

#### How to Test

You can either run:
- bash 
```./scripts/generate_new_api_version.sh 5 6 7```
- gradle task: 
```./gradlew :network:generateNewApiVersion -PpreviousApiVersion=5 -PcurrentApiVersion=6 -PnewApiVersion=7```

It should generate NetworkContainers and API classes.
It should add the new api as BackendMetaDataUtil.DevelopmentApiVersions
There could be some unused params copied from previous API version that developer will need to remove, but static analysis will catch it.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
